### ISSUE TITLE
Add support for Azure.Storage.Blobs in ProtectKeysWithAzureKeyVault

### ISSUE THREAD

**Issue by sopelt** [dotnet/aspnetcore#17366]
We are currently using 
```services
    .AddDataProtection()
    .PersistKeysToAzureBlobStorage(new Uri(blobUri))
    .ProtectKeysWithAzureKeyVault(kv, keyVaultUri);
```
to enable scenarios like scale out and deployment swapping.

Currently [PersistKeysToAzureBlobStorage](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.azuredataprotectionbuilderextensions.persistkeystoazureblobstorage?view=aspnetcore-3.0) and its package [Microsoft.AspNetCore.DataProtection.AzureStorage](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection.AzureStorage/3.1.0-preview3.19555.2) depend on [Microsoft.Azure.Storage.Blob](https://www.nuget.org/packages/Microsoft.Azure.Storage.Blob/). This aspect could also be implemented using [Azure.Storage.Blobs](https://www.nuget.org/packages/Azure.Storage.Blobs/).

In a similar way [ProtectKeysWithAzureKeyVault](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.azuredataprotectionbuilderextensions.protectkeyswithazurekeyvault?view=aspnetcore-3.0#Microsoft_AspNetCore_DataProtection_AzureDataProtectionBuilderExtensions_ProtectKeysWithAzureKeyVault_Microsoft_AspNetCore_DataProtection_IDataProtectionBuilder_Microsoft_Azure_KeyVault_KeyVaultClient_System_String_) and its package [Microsoft.AspNetCore.DataProtection.AzureKeyVault](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection.AzureKeyVault/3.1.0-preview3.19555.2) depend on [Microsoft.Azure.KeyVault](https://www.nuget.org/packages/Microsoft.Azure.KeyVault/). This aspect could also be implemented using [Azure.Security.KeyVault.Keys](https://www.nuget.org/packages/Azure.Security.KeyVault.Keys/).

As we understand the communication/versioning scheme these new libraries seem to be the way forward.

We would like to move to the new blob storage and key vault clients as well as [Azure.Identity](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity) for all our MSI handling and would be happy if the data protection extensions could support that as well. Especially for new applications we would like to avoid the dependency on the "old" libraries.

**Comment by blowdart** [comment]
Do you mean you want the data protection key ring totally in key vault? If so, no. Granting the ability to create keys in key vault to an app violates least privilege and it's api surface doesn't allow for the loading of multiple keys in a single operation, rather you have to page through each key, which would delay app startup by what we feel is an unacceptable time.

**Comment by sopelt** [comment]
Thank you for the swift reply @blowdart ! Sorry for the misunderstanding. That is not what I was asking for. I am aware that this wouldn't match the overall architecture and I agree that write permissions for apps/MSI are to be avoided at all cost.

I tried to clarify and add more information to my initial post.

**Comment by blowdart** [comment]
Oh just an update to the libraries we depend on? Yea, we can look at that for v5.

**Comment by blowdart** [comment]
https://github.com/dotnet/aspnetcore/pull/18203

**Comment by blowdart** [comment]
Closing as done

**Comment by sopelt** [comment]
@blowdart could you please clarify how this will be shipped? The PR references 3.1(.3) but I was not able to find a package/version which doesn't depend on the "old" storage lib. I was under the impression that this will be for 5.x but that label got removed.
Thanks!

**Comment by BrennanConroy** [comment]
The new packages are released and now owned by the Azure SDK team. See this announcement for details: https://github.com/aspnet/Announcements/issues/409
